week3/Program.cs

```diff
@@ -0,0 +1,99 @@
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+app.MapGet("/interface", () =>
+{
+    var cat = new Cat();
+    var dog = new Dog();
+    var cow = new Cow();
+    var animal = new Animal();
+    return animal.MakeSound(cat);
+});
+
+app.MapGet("/account", () =>
+{
+    var account = new Account(200);
+    account.Withdraw(100);
+    account.Deposit(50);
+    return account.Balance;
+});
+
+app.Run();
+
+class Account
+{
+    public int Balance
+    {
+        get
+        {
+            int balance = 0;
+            foreach (var item in transactions)
+            {
+                balance += item.Amount;
+            }
+            return balance;
+        }
+    }
+    public Account(int amount)
+    {
+        Deposit(amount);
+    }
+
+    public List<Transaction> transactions = new List<Transaction>();
```

Comment:
I'm just curious why did you decide to have a list of transactions?

Reply:
I could also create a private field `_balance`, return it by get method and then add or subtract from it by withdraw and deposit methods. It works and  balance is still read only. But I chose to create a list of objects from withdraw or deposit and then loop over them to get balance. According to exercise description, It is better to choose the first solution.